Flaky test "TestImageUpdateAutomationReconciler_Reconcile/no_marker_no_update"

That test just failed in CI and re-running it fixed it:

```
2025-05-04T08:09:13.1513489Z --- FAIL: TestImageUpdateAutomationReconciler_Reconcile (9.66s)
2025-05-04T08:09:13.1514026Z     --- FAIL: TestImageUpdateAutomationReconciler_Reconcile/no_marker_no_update (1.29s)
2025-05-04T08:09:13.1514547Z         imageupdateautomation_controller_test.go:467: 
2025-05-04T08:09:13.1514873Z             Timed out after 1.001s.
2025-05-04T08:09:13.1516330Z             The function passed to Eventually failed at /home/runner/work/image-automation-controller/image-automation-controller/internal/controller/imageupdateautomation_controller_test.go:465 with:
2025-05-04T08:09:13.1517091Z             Expected
2025-05-04T08:09:13.1517318Z                 <int64>: -1
2025-05-04T08:09:13.1517530Z             to equal
2025-05-04T08:09:13.1517747Z                 <int64>: 1
```

[job-logs.txt](https://github.com/user-attachments/files/20034399/job-logs.txt)

--- a/internal/controller/imageupdateautomation_controller_test.go
+++ b/internal/controller/imageupdateautomation_controller_test.go
@@ -308,7 +308,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 		g.Eventually(func(g Gomega) {
 			g.Expect(testEnv.Get(ctx, client.ObjectKeyFromObject(obj), obj)).To(Succeed())
 			g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-		}).Should(Succeed())
+		}, timeout).Should(Succeed())
 
 		// Check if the object status is valid.
 		condns := &conditionscheck.Conditions{NegativePolarity: imageUpdateAutomationNegativeConditions}
@@ -349,7 +349,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 		g.Eventually(func(g Gomega) {
 			g.Expect(testEnv.Get(ctx, client.ObjectKeyFromObject(obj), obj)).To(Succeed())
 			g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-		}).Should(Succeed())
+		}, timeout).Should(Succeed())
 
 		// Check if the object status is valid.
 		condns := &conditionscheck.Conditions{NegativePolarity: imageUpdateAutomationNegativeConditions}
@@ -392,7 +392,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 		g.Eventually(func(g Gomega) {
 			g.Expect(testEnv.Get(ctx, client.ObjectKeyFromObject(obj), obj)).To(Succeed())
 			g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-		}).Should(Succeed())
+		}, timeout).Should(Succeed())
 
 		// Check if the object status is valid.
 		condns := &conditionscheck.Conditions{NegativePolarity: imageUpdateAutomationNegativeConditions}
@@ -428,7 +428,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 				g.Eventually(func(g Gomega) {
 					g.Expect(testEnv.Get(ctx, objKey, &obj)).To(Succeed())
 					g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-				}).Should(Succeed())
+				}, timeout).Should(Succeed())
 
 				// Check if the object status is valid.
 				condns := &conditionscheck.Conditions{NegativePolarity: imageUpdateAutomationNegativeConditions}
@@ -465,7 +465,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 					g.Expect(testEnv.Get(ctx, objKey, &obj)).To(Succeed())
 					g.Expect(obj.Status.ObservedGeneration).To(Equal(obj.GetGeneration()))
 					g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-				}).Should(Succeed())
+				}, timeout).Should(Succeed())
 
 				g.Expect(obj.Status.LastPushCommit).To(BeEmpty())
 				g.Expect(obj.Status.LastPushTime).To(BeNil())
@@ -517,7 +517,7 @@ func TestImageUpdateAutomationReconciler_Reconcile(t *testing.T) {
 					g.Expect(testEnv.Get(ctx, objKey, &obj)).To(Succeed())
 					g.Expect(obj.Status.ObservedGeneration).To(Equal(obj.GetGeneration()))
 					g.Expect(obj.Status.Conditions).To(conditions.MatchConditions(expectedConditions))
-				}).Should(Succeed())
+				}, timeout).Should(Succeed())
 				g.Expect(obj.Status.LastPushCommit).ToNot(BeEmpty())
 				g.Expect(obj.Status.LastPushTime).ToNot(BeNil())
 				g.Expect(obj.Status.LastAutomationRunTime).ToNot(BeNil())

